tls/s2n_handshake_io.c

```diff
@@ -305,6 +305,23 @@ int s2n_conn_set_handshake_type(struct s2n_connection *conn)
     /* A handshake type has been negotiated */
     conn->handshake.handshake_type = NEGOTIATED;
 
+    if (conn->config->use_tickets) {
+        /* First we attempt to resume with session tickets, then we fallback
+         * to attempting to resume with session caching (if enabled).
```

Comment:
My understanding of RFC is that session ticket and session id resumption are mutually exclusive: if client sends a ticket, then it will generate a Session ID, so we should not be able to find it in our cache. So we probably don't even need to try to do a lookup in Session ID cache if client send a ticket to us.

Reply:
> if client sends a ticket, then it will generate a Session ID, so we should not be able to find it in our cache. 

That is only if the client ticket is valid right? 
We proceed to do a lookup in Session ID cache if client sends empty session ticket or when s2n_decrypt_session_ticket fails because the ticket has garbage value.